### ISSUE TITLE
error while specifying variables in CSS3 ; Expected RBRACE. This rule looks for recoverable syntax error

### ISSUE THREAD

**Issue by avreddy1996** [CSSLint/csslint#720]
`:root{
  --nvctheme-color: #3f1a67;
  --nvcthemelight-color: #5f259e;
`
this is the code I am using to define website theme color as a variable but in CSSlint it is showing error
`ERROR: Expected RBRACE at line 17, col 3. This rule looks for recoverable syntax errors. (errors) Browsers: All
`

**Comment by yyualice** [comment]
Experiencing this as well.

**Comment by adityarao310** [comment]
Yup, same here

**Comment by frvge** [comment]
The parser doesn't know how to handle CSS Variables as far as I know.

**Comment by varunyellina** [comment]
Happens with me too.

**Comment by Redmega** [comment]
Support for css variables has landed in all major browsers except ie11. `csslint` should support it!

**Comment by JoniVR** [comment]
@vrosu Nope, same issue as described above, really annoying. Ignoring doesn't seem to help.

**Comment by rhill1030** [comment]
Ran into this issue this morning, Visual Studio won't publish css with this error, even though the pages run just fine locally.  

This is a 6 month old bug, and it hasn't been addressed. If it is too hard to properly write rules to properly identify this use case, then turn off validation checking for empty rules so you don't break things while you get your logic sorted out. 

Kinda irritated that I have to fight with csslint to get work done, it is costing more time than it is saving...

**Comment by LinearDynamics** [comment]
landed here from a search. Have a WordPress site going and the editor didn't like my variable (expected RBRACE).

**Comment by BrandonDyer64** [comment]
Is this still an issue?

**Comment by hroger1030** [comment]
yep. Here is lines 1-8 of a css sheet:

:root
{
    --accent1: #777777;
    --accent2: #cccccc;
    --Background1: #000000;
    --Background2: #000044;
    --Background3: #000055;
}

gives the following errors:

Severity	Code	Description	Project	File	Line	Suppression State
Error	empty-rules	(CssLint) Rule is empty.	Website	D:\SourceCode\test\C# - gc\Website\App_Themes\PageStyle.css	1	
Error	errors	(CssLint) Expected RBRACE at line 3, col 5.	Website	D:\SourceCode\test\C# - gc\Website\App_Themes\PageStyle.css	3	
Error	errors	(CssLint) Expected RBRACE at line 4, col 5.	Website	D:\SourceCode\test\C# - gc\Website\App_Themes\PageStyle.css	4	
Error	errors	(CssLint) Expected RBRACE at line 5, col 5.	Website	D:\SourceCode\test\C# - gc\Website\App_Themes\PageStyle.css	5	
Error	errors	(CssLint) Expected RBRACE at line 6, col 5.	Website	D:\SourceCode\test\C# - gc\Website\App_Themes\PageStyle.css	6	
Error	errors	(CssLint) Expected RBRACE at line 7, col 5.	Website	D:\SourceCode\test\C# - gc\Website\App_Themes\PageStyle.css	7

**Comment by parak80** [comment]
I have the same problem, it gives no error in appveyor client test but in visual studio where my backend included I see errors like above for my variables.css 
:root{
--accent-color: #5491db;
--nav-back-color:#353b44;
...
}
Anyone can help please?

**Comment by BrandonDyer64** [comment]
@parak80 There's not really anything you can do. It's an issue with CSSLint. This issue has been open for over a year.

Speaking of which, are there any Pull Requests addressing this issue?

**Comment by frvge** [comment]
https://github.com/CSSLint/parser-lib/pull/248 got opened a few days ago.

**Comment by shaunrashid** [comment]
Looks like the parser-lib PR is in limbo as the contributor is not working on it anymore. :(

**Comment by EliasTouil** [comment]
Anything new on this ?

**Comment by mattiacci** [comment]
I've opened a pull request to fix this, in case we want to avoid #754.

**Comment by frvge** [comment]
I'm happy to merge it, but I don't have the power to make an official release for neither parser-lib nor csslint itself.

**Comment by 4ist** [comment]
For what it's worth- If my .css files is closed then the .net solution is buildable without the errors

**Comment by tahaygun** [comment]
WTF!!! We are living in 2019 and still struggling with this? I hate visual studio.

**Comment by bakcxoj** [comment]
@tahaygun I am equally frustrated, but I don't think this has anything to do with Visual Studio.

**Comment by tahaygun** [comment]
@BrandonDyer64 probably you are right. I jumped from vscode to visual studio and really hated it.. So I can't afford anything about it. I go crazy every time when I face with a problem.

**Comment by BrandonWingerAir** [comment]
Also got this error with a WordPress site.

**Comment by babakmomeni** [comment]
@BrandonWingerAir 
Could you get rid of this error? I also got this error is my CSS files within WordPress...

**Comment by codler** [comment]
> I'm happy to merge it, but I don't have the power to make an official release for neither parser-lib nor csslint itself.

@nschonni Can you make a release?

**Comment by Ademord** [comment]
I'm still confused and in need of help for this

**Comment by cawa-93** [comment]
@Ademord  I have no hope of solving this problem. The last significant commit was made in 2017. Everything looks like the project is abandoned and dead.

**Comment by sidvenu** [comment]
Codacy, which I think uses CSSLint, has this issue too.

**Comment by oluwafemi-love** [comment]
I am facing this as well.
Probably, I will turn off the CSS Lint.
I am tired.

**Comment by bb010g** [comment]
@nschonni Pinging you, as you're listed in the @CSSLint organization and actively contributing to other linters. Do you know CSSLint's current maintenance status?

**Comment by Xmetalfanx** [comment]
yeah, using Codacy here for a few projects and one of the few reasons it gets rated lower is due to this issue ... I REALLY didn't expect the issue to be this old and still going

**Comment by circuitsacul** [comment]
2021 and this issue is still an issue?

**Comment by thenexus00** [comment]
> 2021 and this issue is still an issue?

Yes, Crazy hey.

**Comment by GeneralMine** [comment]
Can confirm that's still an issue...

**Comment by DevWeb13** [comment]
26/10/2021 Always the issue in codacy...

**Comment by tomivm** [comment]
15/12/2021 the issue persists in codacy

**Comment by S0AndS0** [comment]
For those that land here after a web-search, the following Awk/Bash wrapper script is what I currently use to ignore `RBRACE` errors related to CSS custom properties;

```bash
#!/usr/bin/env bash

_args=( "${@?No arguments provided}" );

awk 'BEGIN {
  delete buffer;
  after_match = 1;
  header_amount = 5;

  pattern_one = "RBRACE";
  pattern_two = ".*--\\w(-|\\w)+:";

  matched_one = 0;
  counter = 0;
}
{
  ## Print header
  if (NR < header_amount) {
    print $0;
    next;
  }

  if (!matched_one) {
    if ($0 ~ pattern_one) {
      # print "## MATCHED PATTERN ONE";
      delete buffer;
      matched_one = 1;
    } else {
      buffer[length(buffer)] = $0;
      if (!length($0)) {
        for (i in buffer) {
          print buffer[i];
        }
        delete buffer;
      }
    }
  } else {
    if ($0 ~ pattern_two) {
      counter++;
    } else if (counter > after_match) {
      counter = 0;
      matched_one = 0;
    }
  }
}' < <(csslint "${_args[@]}");
```

**Comment by chris-oliver-getvisibility** [comment]
It's clear that this repository has been abandoned.

Does anybody know if there is a spiritual successor?

**Comment by S0AndS0** [comment]
@chris-oliver-getvisibility recently I went on the hunt for alternatives, and it seems [stylelint](https://github.com/stylelint/stylelint) may be able to satisfy the want of CSS (and SCSS) linting... though I've yet to make the time to test it :shrug:

**Comment by chris-oliver-getvisibility** [comment]
Thanks @S0AndS0 !

**Comment by linusjf** [comment]
Any updates on this? CSS variables are breaking csslint.

**Comment by linusjf** [comment]
Dropped the use of csslint from my project. stylelint seems to be the way forward.

**Comment by Xmetalfanx** [comment]
> Dropped the use of csslint from my project. stylelint seems to be the way forward.

i am not even sure (it was at LEAST MONTHS ago or longer) but i think that is what i did here
